docs/book/helpers/doctype.md

```diff
@@ -26,39 +26,58 @@ You can also specify a custom doctype as long as it is well-formed.
 The `Doctype` helper is a concrete implementation of the
 [Placeholder helper](placeholder.md).
 
-## Basic Usage
+## Configuration and Basic Usage
 
 You may specify the doctype at any time. However, helpers that depend on the
-doctype for their output will recognize it only after you have set it, so the
-easiest approach is to specify it in your bootstrap:
+doctype for their output will recognize it only after you have set it.
+There are multiple ways to set the doctype:
 
-```php
-use Laminas\View\Helper\Doctype;
+### Generic application configuration
 
-$doctypeHelper = new Doctype();
-$doctypeHelper->doctype('XHTML1_STRICT');
-```
-
-And then print it out on top of your layout script:
+`\Laminas\View\Helper\Service\DoctypeFactory` checks the application configuration, making it possible to
+define the doctype through your configuration (e.g. a `ConfigProvider`):
 
 ```php
-<?php echo $this->doctype() ?>
+return [
+    'view_helper_config' => [
+        'doctype' => 'HTML5',
+    ],
+];
 ```
 
-Within an application based off the [laminas-mvc-skeleton](https://github.com/laminas/laminas-mvc-skeleton),
-you can specify the doctype via configuration:
+### `laminas-mvc` configuration
+
+An application based off the [laminas-mvc-skeleton](https://github.com/laminas/laminas-mvc-skeleton)
+comes with its own factory in the `ViewHelperManager` and can be configured like this:
 
 ```php
 // module/Application/config/module.config.php:
 return [
     /* ... */
     'view_manager' => [
-        'doctype' => 'html5',
+        'doctype' => 'HTML5',
```

Comment:
When writing the docs I noticed that the doctype has to be in uppercase, even in `laminas-mvc`.

@weierophinney @froschdesign Is it okay if I change that here or do you want me to change that in a separate PR? Or maybe even change `Doctype` itself to accept lowercase?

Reply:
@rieschl
> Is it okay if I change that here…

Using a separate commit on this pull request is enough. Btw. good catch! 👍 

> Or maybe even change Doctype itself to accept lowercase?

Good idea! Create an issue report or a pull request – both are welcome.

Reply:
Thanks for the quick reply!


> @rieschl
> > Is it okay if I change that here…
> 
> Using a separate commit on this pull request is enough.

Will do!

> Btw. good catch! 👍 

Thanks. finding and fixing bugs, code reviewing and keeping code quality (tools) up to date are my main tasks in my biggest project, so I kind of have an eye for things like that 🙂


> 
> > Or maybe even change Doctype itself to accept lowercase?
> 
> Good idea! Create an issue report or a pull request – both are welcome.
> 

okay I'll come up with a PR, but probably not today anymore.

Reply:
> 
> Using a separate commit on this pull request is enough.

Done :)